### PR TITLE
fix: docker-compose file decoding

### PR DESCRIPTION
**Pull request by n1ru4l**
Closes https://github.com/erikengervall/dockest/issues/183

### CODE REVIEW

File: packages/examples/aws-codebuild/src/CI.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM node:12-slim
```
Comment (n1ru4l) [inline packages/examples/aws-codebuild/src/CI.Dockerfile:1]: I copied the image inside this repository. This makes it easier to follow whats going on.

File: packages/examples/aws-codebuild/src/buildspec.yml [hunk 2]
```diff
@@ -5,8 +5,8 @@ phases:
     commands:
       # docker in docker integration
       - nohup /usr/bin/dockerd --host=unix:///var/run/docker.sock --host=tcp://127.0.0.1:2375 --storage-driver=overlay2&
-      - timeout -t 15 sh -c "until docker info; do echo .; sleep 1; done"
-      - yarn
+      - timeout 15 sh -c "until docker info; do echo .; sleep 1; done"
+      - yarn install --no-lockfile
```
Comment (n1ru4l) [inline packages/examples/aws-codebuild/src/buildspec.yml:9]: This caused a lot of issues with outdated lock files. I disabled it for now

File: packages/examples/aws-codebuild/src/codebuild_test.sh [hunk 3]
```diff
@@ -4,6 +4,8 @@ set -euxo pipefail
 echo "Run dockest as usual"
 yarn test:codebuild:buildspec
 
-# image source: https://github.com/n1ru4l/docker-image-node-10-with-docker-and-compose
+echo "Build docker image"
+docker build --file CI.Dockerfile -t aws-codebuild-ci-node .
```
Comment (n1ru4l) [inline packages/examples/aws-codebuild/src/codebuild_test.sh:8]: As mentioned above we now use the image inside this repo :) I do not intend to maintain the other one as we are using a custom one at work anyways. So we can maintain a version co-located with dockest that can be adjusted when tests fail/the image becomes outdated.

**Comment by n1ru4l** [comment]
@erikengervall Let me know what you think :)

**Comment by n1ru4l** [comment]
@erikengervall ping :)

**Comment by erikengervall** [comment]
> @erikengervall Let me know what you think :)

Sorry for the late reply - been a busy few weeks 🙂 

I'll take a look tonight! 🙌

**Comment by erikengervall** [comment]
Ran the unit tests as a sanity check and it produces these changes for me locally:

Screenshot 2020-10-12 at 17 48 58

I'm running docker desktop 2.4.0.0

Perhaps we should update the snapshots to the most recent version? WDYT?

**Comment by n1ru4l** [comment]
@erikengervall yes! I actually thought that I did update them already 😅

**Comment by n1ru4l** [comment]
It seems like the port format got reverted to in 1.27.4: https://github.com/docker/compose/compare/1.27.3...1.27.4 (https://docs.docker.com/compose/release-notes/#1274)

I would stay we should steel keep this, as it gives another layer of confidence

**Review by erikengervall — APPROVED** [review]

**Comment by n1ru4l** [comment]
@erikengervall Any ideas what's wrong with the unit tests? Some checks are not reported but required? I would love to see this merged quickly! Gonna pick up and finish https://github.com/erikengervall/dockest/pull/167 afterwards 🙂

**Comment by erikengervall** [comment]
> @erikengervall Any ideas what's wrong with the unit tests? Some checks are not reported but required? I would love to see this merged quickly! Gonna pick up and finish https://github.com/erikengervall/dockest/pull/167 afterwards 🙂

I'll take a look when I get home later today :)

**Comment by erikengervall** [comment]
> > @erikengervall Any ideas what's wrong with the unit tests? Some checks are not reported but required? I would love to see this merged quickly! Gonna pick up and finish #167 afterwards 🙂
> 
> I'll take a look when I get home later today :)

Alright idk why GitHub Actions are acting up, the tests are seemingly passing and I've tried running them locally with the most recent changes.

I'll merge this and create a new release 🙂

**Comment by erikengervall** [comment]
@n1ru4l changes included in [2.1.0](https://www.npmjs.com/package/dockest/v/2.1.0) 🙂

**Pull request merged**